### PR TITLE
Bump gateway-api to v1.1.0

### PR DESCRIPTION
**Pull request by sunjayBhatia**
Fixes: https://github.com/projectcontour/contour/issues/6397

### CODE REVIEW

**Comment by sunjayBhatia** [comment]
Looks like `HTTPRouteHTTPSListener` is failing, it expects a 404 for request to a HTTPS listener with an unknown hostname

Seems like we should either push back on this (expect a TLS error instead) or we'll have to ensure a fallback certificate is configured and enabled so we can terminate TLS and return an HTTP response

**Comment by sunjayBhatia** [comment]
> Looks like `HTTPRouteHTTPSListener` is failing, it expects a 404 for request to a HTTPS listener with an unknown hostname
> 
> Seems like we should either push back on this (expect a TLS error instead) or we'll have to ensure a fallback certificate is configured and enabled so we can terminate TLS and return an HTTP response

hm looks like i approved this test and it passed before, will have to take another look into why its failing in CI

**Comment by sunjayBhatia** [comment]
> > Looks like `HTTPRouteHTTPSListener` is failing, it expects a 404 for request to a HTTPS listener with an unknown hostname
> > Seems like we should either push back on this (expect a TLS error instead) or we'll have to ensure a fallback certificate is configured and enabled so we can terminate TLS and return an HTTP response
> 
> hm looks like i approved this test and it passed before, will have to take another look into why its failing in CI

looks like i approved before the negative hostname test was added and didnt re-check it later

**Comment by sunjayBhatia** [comment]
> > > Looks like `HTTPRouteHTTPSListener` is failing, it expects a 404 for request to a HTTPS listener with an unknown hostname
> > > Seems like we should either push back on this (expect a TLS error instead) or we'll have to ensure a fallback certificate is configured and enabled so we can terminate TLS and return an HTTP response
> > 
> > 
> > hm looks like i approved this test and it passed before, will have to take another look into why its failing in CI
> 
> looks like i approved before the negative hostname test was added and didnt re-check it later
> 
> opened [kubernetes-sigs/gateway-api#3044](https://github.com/kubernetes-sigs/gateway-api/issues/3044)

realized this is probably a contour deficiency, I don't think we generate a filter chain if there is no httproute attached to a listener

**Comment by sunjayBhatia** [comment]
upgrade test probably failing since we started reconciling v1alpha3 backendtlspolicy and that has not been applied to the cluster as part of the upgrade

**Comment by sunjayBhatia** [comment]
Listener isolation test failing, needs this fix: https://github.com/projectcontour/contour/pull/6162

**Comment by sunjayBhatia** [comment]
disabled backendtlspolicy feature in provisioner upgrade test since the upgrade from v1.0.0 to v1.1.0 of gw api removes the old api version and adds v1alpha3 of the resource

i tried instead to install the newer crds as part of the upgrade but the grpcroute crd going to v1 also causes some issues, could be more surgical but this felt easier for now

**Comment by sunjayBhatia** [comment]
> disabled backendtlspolicy feature in provisioner upgrade test since the upgrade from v1.0.0 to v1.1.0 of gw api removes the old api version and adds v1alpha3 of the resource
> 
> i tried instead to install the newer crds as part of the upgrade but the grpcroute crd going to v1 also causes some issues, could be more surgical but this felt easier for now

nvm, just fixed the crd install as part of the upgrade

found a bug when setting multiple disabled features in the provisioner

File: internal/gatewayapi/helpers.go [hunk 1]
```diff
@@ -149,36 +149,36 @@ func TLSRouteBackendRef(serviceName string, port int, weight *int32) []gatewayap
...
+func GRPCRouteBackendRef(serviceName string, port int, weight int32) []gatewayapi_v1.GRPCBackendRef {
+	return []gatewayapi_v1.GRPCBackendRef{
 		{
-			BackendRef: gatewayapi_v1alpha2.BackendRef{
-				BackendObjectReference: gatewayapi_v1alpha2.BackendObjectReference{
+			BackendRef: gatewayapi_v1.BackendRef{
+				BackendObjectReference: gatewayapi_v1.BackendObjectReference{
 					Group: ptr.To(gatewayapi_v1.Group("")),
 					Kind:  ptr.To(gatewayapi_v1.Kind("Service")),
 					Name:  gatewayapi_v1alpha2.ObjectName(serviceName),
```
Comment (skriss) [inline internal/gatewayapi/helpers.go:159]: nit, looks like this line could be switched to use v1 as well

File: internal/gatewayapi/helpers.go [hunk 2]
```diff
@@ -126,19 +126,19 @@ func HTTPBackendRef(serviceName string, port int, weight int32) []gatewayapi_v1.
...
-func TLSRouteBackendRef(serviceName string, port int, weight *int32) []gatewayapi_v1alpha2.BackendRef {
-	return []gatewayapi_v1alpha2.BackendRef{
+func TLSRouteBackendRef(serviceName string, port int, weight *int32) []gatewayapi_v1.BackendRef {
+	return []gatewayapi_v1.BackendRef{
 		{
-			BackendObjectReference: gatewayapi_v1alpha2.BackendObjectReference{
+			BackendObjectReference: gatewayapi_v1.BackendObjectReference{
 				Group: ptr.To(gatewayapi_v1.Group("")),
 				Kind:  ptr.To(gatewayapi_v1.Kind("Service")),
 				Name:  gatewayapi_v1alpha2.ObjectName(serviceName),
```
Comment (skriss) [inline internal/gatewayapi/helpers.go:144]: nit, looks like this line could be switched to use v1 as well

File: test/conformance/gatewayapi/gateway_conformance_test.go [hunk 3]
```diff
@@ -79,15 +84,38 @@ func TestGatewayConformance(t *testing.T) {
 			// implemented plus complications with the test setup itself.
 			// See: https://github.com/projectcontour/contour/issues/5922
 			tests.GatewayStaticAddresses.ShortName,
+
+			// Skip this test as it uses a Gateway with a name that is too long,
+			// adding the name in a label value prevents resources to be
+			// created.
+			tests.HTTPRouteInvalidParentRefSectionNameNotMatchingPort.ShortName,
```
Comment (skriss) [inline test/conformance/gatewayapi/gateway_conformance_test.go:91]: 🤮 would be nice to change this upstream if possible, have to imagine other folks will run into issues too

**Review by skriss — COMMENTED** [review]
Looks pretty good, just a couple nits so far, still need to review the BackendTLSPolicy changes in a little more detail but not too concerned about it.

File: test/conformance/gatewayapi/gateway_conformance_test.go (hunk 3 above)
Comment (skriss) [reply test/conformance/gatewayapi/gateway_conformance_test.go:91]: (vomit face was for the upstream test using too long a name, not you disabling it!)

Comment (sunjayBhatia) [reply test/conformance/gatewayapi/gateway_conformance_test.go:91]: ha yeah i figured, and agreed tbh

I think there was a discussion on this upstream i'll have to find it again

File: changelogs/unreleased/6398-sunjayBhatia-minor.md [hunk 4]
```diff
@@ -0,0 +1,9 @@
+## Update to Gateway API v1.1.0
+
+Gateway API CRD compatibility has been update to release v1.1.0.
```
Comment (skriss) [inline changelogs/unreleased/6398-sunjayBhatia-minor.md:3]:
```suggestion
Gateway API CRD compatibility has been updated to release v1.1.0.
```

File: internal/dag/cache.go [hunk 5]
```diff
@@ -856,45 +857,40 @@ func (kc *KubernetesCache) LookupService(meta types.NamespacedName, port intstr.
 	return nil, core_v1.ServicePort{}, fmt.Errorf("port %q on service %q not matched", port.String(), meta)
 }
 
-// LookupBackendTLSPolicyByTargetRef returns the Kubernetes BackendTLSPolicies that matches the provided targetRef with
+// LookupBackendTLSPolicyByTargetRef returns the Kubernetes BackendTLSPoliciy that matches the provided targetRef with
```
Comment (skriss) [inline internal/dag/cache.go:860]:
```suggestion
// LookupBackendTLSPolicyByTargetRef returns the Kubernetes BackendTLSPolicy that matches the provided targetRef with
```

**Review by skriss — APPROVED** [review]
LGTM, just a couple typos

**Pull request merged**